client/http4s-client/src/main/scala/sttp/tapir/client/http4s/Http4sClientOptions.scala

```diff
@@ -1,10 +1,9 @@
 package sttp.tapir.client.http4s
 
 import sttp.tapir.Defaults
+import sttp.tapir.internal.TapirFile
 
-import java.io.File
-
-case class Http4sClientOptions(createFile: () => File)
+case class Http4sClientOptions(createFile: () => TapirFile)
```

Comment:
maybe in fact we need both: a `sttp.tapir.File` alias which allows to cross-compile code. And a `TapirFile` internal-only case class which represents files with additional meta-data. The `TapirFile` could then implement a platform-specific `.from(sttp.tapir.File): TapirFile` method. This would also simplify testing

Reply:
or yet another approach. We won't manage to keep `TapirFile` in `internal`, as it leaks to non-internal APIs in a number of places. So let's make it public.  And maybe let's call it what it is: a `FileRange`, where it can be the entire file, or a byte range.

For user convenience, we can define the `sttp.tapir.File` alias (as described above). But the low-level representation, to which codecs encode files will be `FileRange`; both in the public API. What do you think?

Reply:
Second idea sounds better to me

Reply:
let's do it then :)